enginetest/trigger_queries.go

```diff
@@ -1864,6 +1864,739 @@ end;`,
 	},
 }
 
+// RollbackTriggerTests are trigger tests that require rollback logic to work correctly
+var RollbackTriggerTests = []ScriptTest{
```

Comment:
These are great, nice work.

You also need an additional set of tests that exercise transaction logic. They should set @@autocommit = off, then do a bunch of work in the same transaction, the last statement of which gets rolled back